OvmfPkg/Library/PlatformInitLib/Platform.c

```diff
@@ -555,6 +556,26 @@ PlatformMaxCpuCountInitialization (
   IN OUT EFI_HOB_PLATFORM_INFO  *PlatformInfoHob
   )
 {
+ #ifdef MDE_CPU_X64

```

Comment:
This #ifdef indicates that some abstractions are lacking here. This logic should probably be in a support library, and the default, non-TDVF versions should just do nothing.

Reply:
Currently, TdxLib that contain TdInfo is only built for X64 in PlatformInitLib.inf and TdxLib is not defined in OvmfPkgIa32.dsc.
So , there is a build error if no #ifdef.

Now , add the TdxLib definition in OvmfPlgIa32.dsc and change TdxLib to common in PlatformInitLib.inf,  
then we can remove the #ifdef